app/models/order.rb

```diff
@@ -12,7 +12,7 @@ def grandtotal
   end
 
   def merchant_grandtotal(id)
-    items.where(merchant_id: id).sum('items.price * quantity')
+    items.where(merchant_id: id).sum('item_orders.price * item_orders.quantity')
```

Comment:
used the item orders price instead of the items price and this handles discounts and non discounts